src/conversion.rs

```diff
@@ -641,4 +656,22 @@ mod tests {
             assert_eq!(none.get_refcnt(), ref_cnt);
         });
     }
+
+    #[test]
+    fn test_box_topyobject() {
+        let s: Box<str> = "test".into();
+
+        let obj = Python::with_gil(|py| s.to_object(py));
+
+        assert_eq!(&obj.to_string(), "test")
+    }
+
+    #[test]
+    fn test_box_intopy() {
+        let s: Box<str> = "test".into();
+
+        let obj: PyObject = Python::with_gil(|py| s.into_py(py));
```

Comment:
The fact that I need to specifically annotate the type of `obj` here makes me suspicious that implementing `IntoPy` like above is doing something funky.